### PR TITLE
Integration test to check that the ORM definitions match the LORIS SQL database

### PR DESCRIPTION
**Pull request by maximemulder**
# Description of the PR

This PR adds an integration test that checks that the ORM definitions correspond to the SQL database as defined in the main LORIS repository.

It notably checks  the following:
- Check that each ORM model has a corresponding table.
- Check that each ORM model has the same columns as its table.
- Check that each ORM column has a similar type as its SQL column (on an abstract level, it does not check for details such as column length like `INT(11)` vs `INTEGER`, it does check for nullability).

To make this test possible, this PR also improves the LORIS-MRI Docker / integration setup, in particular the connection between the database image and the main LORIS-MRI image.

Notably, the Raisinbread database is now built directly into the database image, which allows to cache it with the image and not resource the SQL files each time the database image is ran (which is the default MariaDB Docker behaviour). This should nullify the need for LORIS-MRI Raisinbread overwrites, which could be removed in a follow-up PR. Note however that since both the main LORIS-MRI and database Docker images are quite large, it still takes ~3min to import the cached Docker layers.

There are also a few changes made to the models in this PR. Some are things caught by this PR (whooopsie !), some others are just improvements that this PR makes me confident enough to do.

### CODE REVIEW

**Review by cmadjar — COMMENTED** [review]
Looks good. One tiny question to answer though.

File: python/tests/integration/test_orm_sql_sync.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+            # Check that the types of the ORM and SQL column are equal
+            # The type comparison is not exact, minor differences are ignored
+            sql_column = sql_table.columns[orm_column.name]
+            orm_column_python_type = get_orm_python_type(orm_column.type)
+            sql_column_python_type = get_sql_python_type(sql_column.type)
+            assert orm_column_python_type == sql_column_python_type
+            assert orm_column.nullable == sql_column.nullable
+
+        print()
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply python/tests/integration/test_orm_sql_sync.py:51]: @maximemulder maybe I should have tagged you so you could see the review?

[the remark this answers is not in the document]
Comment (maximemulder) [reply python/tests/integration/test_orm_sql_sync.py:51]: Oh, it's just a statement to add an empty line break between each table in the output, the result looks like this:
```
test table: candidate
    test field: CandID
    test field: PSCID
    ...
    test field: Active
    
test table: session
    test field: ID
    ...
```

Note that the output only appears when the test fails though. I should probably have added a fail commit in the history as a demonstration, I'll do that tomorrow morning (now is not really the time for that 😅).

[the remark this answers is not in the document]
Comment (maximemulder) [reply python/tests/integration/test_orm_sql_sync.py:51]: @cmadjar See the "test error" commit integration test to see what the error looks like if the ORM mappings are wrong.

**Review by cmadjar — APPROVED** [review]
👍

**Pull request merged**